src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -255,49 +327,59 @@ core::smart_refctd_ptr<ICPUShader> CHLSLCompiler::compileToSPIRV(const char* cod
             hlslOptions.preprocessorOptions.logger.log("invalid shader stage %i", system::ILogger::ELL_ERROR, stage);
             return nullptr;
     };
-
-    std::vector<LPCWSTR> arguments = {
-        L"-spirv",
+    
+    std::vector<std::wstring> arguments = {};
+    if (dxc_compile_flags.size()) { // #pragma dxc_compile_flags takes priority
+        populate_arguments_with_type_conversion(arguments, dxc_compile_flags, hlslOptions.preprocessorOptions.logger);
+    }
+    else if (hlslOptions.dxcOptions.size()) { // second in order of priority is command line arguments
+        populate_arguments_with_type_conversion(arguments, hlslOptions.dxcOptions, hlslOptions.preprocessorOptions.logger);
```

Comment:
you could unify both branches if you only initialized `dxc_compile_flags` to `hlslOptions.dxcOptions`